flow/view_holder.cc

```diff
@@ -150,22 +127,47 @@ void ViewHolder::set_hit_testable(bool value) {
 }
 
 void ViewHolder::set_focusable(bool value) {
-  view_properties_.focus_change = value;
+  if (view_properties_.focus_change != value) {
+    view_properties_.focus_change = value;
+    view_properties_changed_ = true;
+  }
 }
 
 void ViewHolder::set_size(const SkSize& size) {
+  if (size.width() > 0.f && size.height() > 0.f) {
+    if (view_properties_.bounding_box.max.x != size.width()) {
+      view_properties_.bounding_box.max.x = size.width();
+      view_properties_changed_ = true;
+    }
+    if (view_properties_.bounding_box.max.y != size.height()) {
+      view_properties_.bounding_box.max.y = size.height();
+      view_properties_changed_ = true;
+    }
+  }
+
   // TODO(dworsham): The Z-bound should be derived from elevation.  We should be
   // able to Z-limit the view's box but otherwise it uses all of the available
   // airspace.
-  view_properties_.bounding_box.max.x = size.width();
-  view_properties_.bounding_box.max.y = size.height();
+  view_properties_.bounding_box.min.z = kArbitraryLargeDepth;
 }
 
 void ViewHolder::set_occlusion_hint(const SkRect& occlusion_hint) {
-  view_properties_.inset_from_min.x = occlusion_hint.fLeft;
-  view_properties_.inset_from_min.y = occlusion_hint.fTop;
-  view_properties_.inset_from_max.x = occlusion_hint.fRight;
-  view_properties_.inset_from_max.y = occlusion_hint.fBottom;
+  if (view_properties_.inset_from_min.x != occlusion_hint.fLeft) {
```

Comment:
Same question here about float equality.

Reply:
It's also OK here.  We want the code to trigger if even the smallest change is detected (so anything larger than 1e-7 for a float)